docs/development/abi.md

```diff
@@ -0,0 +1,219 @@
+# Pyodide Platform ABI
+
+## ABIs
+
+### General
+
+Shared libraries must be linked with `-sSIDE_MODULE=1` or
+
+```
+-sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS=<export list>
+```
+
+Note that the name of each symbol in the list needs to be prefixed with an
+underscode. For the smallest result, it is recommended to link with:
+
+```
+-sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS=["_PyInit_MyCModule1", "_PyInit_MyCModule2]
+```
+
+To force all symbols to be exported, link with `-sSIDE_MODULE=1`.
+
+If `-pthread` is used at compile or link time, the resulting libraries will not
+load.
+
+To compile Rust packages, the following flags must be passed to `rustc`:
+
+```
+-C link-arg=-sSIDE_MODULE=2 -C link-arg=-sWASM_BIGINT
+```
+
+When not invoking `rustc` directly, for instance when using cargo, set these
+into the `RUSTFLAGS` environment variable.
+
+Compiling with `-sSIDE_MODULE=1` will not work with Rust because Rust libraries
+contain a `lib.rmeta` file which is not an object file. Rust produces the
+correct list of exported symbols automatically so this should not be a problem
+in practice.
+
+On Rust libc versions older than 0.2.162, it may be necessary to pass
+`-Z link-native-libraries=no` as a `RUSTFLAG`.
+
+Pyodide is statically linked with the following libraries:
+
+- `libegl.js`
+- `libeventloop.js`
+- `libGL`
+- `libhtml5_webgl.js`
+- `libhtml5.js`
+- `liblz4`
+- `libsdl.js`
+- `libwebgl.js`
+- `libwebsocket.js`
+- `bzip2`
+- `zlib`
```

Comment:
Although it isn't version specific at present, Is this list likely to be version specific? It might be worth flagging that it's explicitly "consistent on all currently ABI tags".

Reply:
Right, it's expected that it will eventually become version-specific and then I'll duplicate this list into each of the specific ABIs.